### ISSUE TITLE
HMAC SHA1

### ISSUE THREAD

**Issue by 99xs** [krzyzanowskim/CryptoSwift#234]
I'm obviously new to this, but maybe you can give me some help on how to do HMAC with CryptoSwift. I know this might be basic stuff for you, and I apologize if my questions seem a bit stupid. :)

I have this code:

    let parameters:String = "\(oauthNonce)&\(oauthSignatureMethod)&\(timestamp)&\(oauthVersion)&\(oauthConsumerKey)&\(oauthToken)"
    let encKey:String = "\(consumerSecretEnc!)&\(tokenSecretEnc!)"

Basically, I need to generate a signature, with HMAC-SHA1 signature method. In the end, I need to base64 encode the hmac so I get my final signature. (similar to base64_encode(hash_hmac('SHA1', $signatureBaseString, $key, 1)) in php).

Any help would be very much appreciated. Thank you so much in advance.

**Comment by krzyzanowskim** [comment]
@99xs start here https://github.com/krzyzanowskim/CryptoSwift#usage hint "Message authenticators"

**Comment by wackymaster** [comment]
@99xs were you able to find a solution to this?
